rust/common-runtime/src/sandbox/wasmtime/compile/module.rs

```diff
@@ -61,8 +61,12 @@ where
     ) -> Result<Self::ModuleInstance, CommonRuntimeError> {
         let mut store = Store::new(&self.engine, ModuleHostState::new(io));
 
-        let (common, instance) = Common::instantiate(&mut store, &self.component, &self.linker)
-            .map_err(|error| CommonRuntimeError::ModuleInstantiationFailed(format!("{error}")))?;
+        let (common, instance) =
+            Common::instantiate_async(&mut store, &self.component, &self.linker)
```

Comment:
Are the async bindings only used here due to the nested Tokio runtimes in the sync wasmtime/wasi bindings? While we won't be able to get away from managing some concept of ["fuel"](https://docs.rs/wasmtime/latest/wasmtime/?search=fuel), keeping host APIs "sync" until we can fully stub tokio usage in wasi (and attempt to move back to sync bindings) sounds like a good plan, unless you have some ideas or plans on async host APIs

Reply:
I don't think it's worth the trouble to stay "sync" when it adds the insane complexity of not inheriting our chosen runtime. AFAICT the async bindings are just as easy to implement, and have the added benefit of enabling us to use an async storage interface at some point.

Reply:
In case it needs to be observed explicitly: we actually cannot use the sync instantiation APIs in, say, any async server (be it Axum, Tonic or most others) short of manually spawning threads. That's not a place we want to be in.

Reply:
A separate observation: we will probably reach a point where we want to spawn the runtime in a separate process (for the sake of memory isolation). At that point, if the async instantiation is still a problem, we will have the option to switch back within significantly changing the complexity of the implementation.

Reply:
> I don't think it's worth the trouble to stay "sync" when it adds the wild complexity of using async internally _and_ not inheriting our chosen runtime. AFAICT the async bindings are just as easy to implement, and have the added benefit of enabling us to use an async storage interface at some point.

I'm suggesting that properly stubbed, we will not need any async runtime in the bindings (neither BYO or wasmtime's ephemeral), unless there's a need for it with the Common components

Reply:
If we used the sync bindings then (in a hypothetical future) accessing an async host API would necessitate a `block_on` in the binding. Ultimately the async bindings are more coherent with where we will be making use of the runtime.

Reply:
Example cases that are very likely and where async bindings will be more coherent:

- We access Web Crypto API or IndexedDB in a host binding
- We offer an API for invoking a recipe from a recipe (and blocking internally until we get a response)